hail/python/hailtop/hailctl/auth/login.py

```diff
@@ -94,9 +34,8 @@ async def auth_flow(deploy_config, default_ns, session):
 
 async def async_login(namespace: Optional[str]):
     deploy_config = get_deploy_config()
-    if namespace:
-        deploy_config = deploy_config.with_default_namespace(namespace)
     namespace = namespace or deploy_config.default_namespace()
-    headers = await hail_credentials(namespace=namespace, authorize_target=False).auth_headers()
-    async with client_session(headers=headers) as session:
-        await auth_flow(deploy_config, namespace, session)
+    async with hail_credentials(namespace=namespace, authorize_target=False) as credentials:
+        headers = await credentials.auth_headers()
+        async with client_session(headers=headers) as session:
```

Comment:
why is this inside the credentials with? Once we have the headers we can close the creds, right?

Reply:
Ya it does not have to be.